tests/Projection/PdoEventStoreProjectorTest.php

```diff
@@ -168,4 +168,35 @@ public function it_throws_exception_when_unknown_event_store_instance_passed():
             10000
         );
     }
+
+    /**
+     * @test
+     */
+    public function it_dispatches_pcntl_signals_when_enabled(): void
+    {
+        $this->prepareEventStream('user-123');
+        $this->connection->exec('DROP TABLE projections;');
+
+        $command = 'php ' . realpath(__DIR__) . '/isolated-projection.php';
+        $descriptorSpec = [
+            0 => ['pipe', 'r'],
+            1 => ['pipe', 'w'],
+            2 => ['pipe', 'w'],
+        ];
+        /**
+         * Created process inherits env variables from this process.
+         * Script returns with non-standard code SIGUSR1 from the handler and -1 else
+         */
+        $projectionProcess = proc_open($command, $descriptorSpec, $pipes);
```

Comment:
This turned out to be harder than I thought and brought me down a road I certainly don't want to head to :) 

Problem with exec is that it waits for the command to finish and doesn't return the pid. We could resolve this by sending the command to the background (`php ... &`) or return the pid (`php ... & echo $!;`). But still then, it doesn't return if the output is not redirected to `/dev/null` (see comment in [documentation](http://php.net/manual/en/function.exec.php)). SO I opted for the `proc*` family instead. 

The test itself is rather simple - we return a non-standard exit code in our callback instead of the default one. Disabling the flag on the projection makes this test fail.